ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java

```diff
@@ -1869,7 +1874,10 @@ public void close() throws IOException {
       unCacheDataNucleusClassLoaders();
     } finally {
       // removes the threadlocal variables, closes underlying HMS connection
-      Hive.closeCurrent();
+      if (hiveDb != null) {
+        hiveDb.close(true);
```

Comment:
That's correct.  The original design set the "allowClose" variable to false since there is cleanup logic when a thread dies that could close the session.  Since this is explicitly called at session close time, we can safely close the connection.